src/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -1203,7 +1203,7 @@ protected void setAttribute(final short tag, final Object value) {
 
     protected void setAttribute(final short tag, final Object value, final boolean isUnsignedArray) {
         if (value != null &&
-                !(value instanceof Byte || value instanceof Short || value instanceof Integer ||
+                !(value instanceof Byte || value instanceof Short || value instanceof Integer || value instanceof Long ||
```

Comment:
I don't think my concerns from last time about allowing `Longs` into `SAMRecord` attributes have been adequately addressed. We need a range check at this `setAttribute()` entry point for the case where `value instanceof Long`, and throw an `IllegalArgumentException` from here if `value < 0 || value > BinaryCodec.MAX_UINT` (plus corresponding unit tests showing that it throws in these cases). I think we should also add a `getUnsignedIntegerAttribute()` accessor to `SAMRecord` that returns a `Long`, since the existing `getIntegerAttribute()` method can't handle the full range of 32-bit unsigned int values.